File: templates/Virtual-Assistant-Template/typescript/samples/sample-assistant/src/responses/cancelResponses.ts
/**
 * Copyright(c) Microsoft Corporation.All rights reserved.
 * Licensed under the MIT License.
 */

import i18next from 'i18next';
import {
    DictionaryRenderer,
    LanguageTemplateDictionary,
    TemplateFunction } from '../templateManager/dictionaryRenderer';
import { TemplateManager } from '../templateManager/templateManager';

export class CancelResponses extends TemplateManager {

    // Declare here the type of properties and the prompts
    public static responseIds: {
        cancelPrompt: string;
        cancelConfirmedMessage: string;
        cancelDeniedMessage: string;
        nothingToCancelMessage: string;
    } = {
        cancelPrompt: 'cancelPrompt',
        cancelConfirmedMessage: 'cancelConfirmed',
        cancelDeniedMessage: 'cancelDenied',
        nothingToCancelMessage: 'nothingToCancel'
    };

    // Declare the responses map prompts
    private static readonly responseTemplates: LanguageTemplateDictionary = new Map([
        ['default', new Map([
            [CancelResponses.responseIds.cancelConfirmedMessage, CancelResponses.fromResources('cancel.cancelConfirmed')],
            [CancelResponses.responseIds.cancelDeniedMessage, CancelResponses.fromResources('cancel.cancelDenied')],
            [CancelResponses.responseIds.cancelPrompt, CancelResponses.fromResources('cancel.cancelPrompt')],
            [CancelResponses.responseIds.nothingToCancelMessage, CancelResponses.fromResources('cancel.nothingToCancel')]
        ])]
    ]);

    // Initialize the responses class properties
    constructor() {
        super();
        this.register(new DictionaryRenderer(CancelResponses.responseTemplates));
    }

    private static fromResources(name: string): TemplateFunction {
        return (): Promise<string> => Promise.resolve(i18next.t(name));
    }
}

File: samples/EnterpriseNotification/VirtualAssistant/Responses/Cancel/CancelResponses.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using Microsoft.Bot.Builder;
using Microsoft.Bot.Builder.TemplateManager;
using Microsoft.Bot.Schema;

namespace VirtualAssistant.Responses.Cancel
{
    public class CancelResponses : TemplateManager
    {
        private static LanguageTemplateDictionary _responseTemplates = new LanguageTemplateDictionary
        {
            ["default"] = new TemplateIdMap
            {
                {
                    ResponseIds.CancelConfirmedMessage,
                    (context, data) =>
                    MessageFactory.Text(
                        text: CancelStrings.CANCEL_CONFIRMED,
                        ssml: CancelStrings.CANCEL_CONFIRMED,
                        inputHint: InputHints.AcceptingInput)
                },
                {
                    ResponseIds.CancelDeniedMessage,
                    (context, data) =>
                    MessageFactory.Text(
                        text: CancelStrings.CANCEL_DENIED,
                        ssml: CancelStrings.CANCEL_DENIED,
                        inputHint: InputHints.AcceptingInput)
                },
                {
                    ResponseIds.CancelPrompt,
                    (context, data) =>
                    MessageFactory.Text(
                        text: CancelStrings.CANCEL_PROMPT,
                        ssml: CancelStrings.CANCEL_PROMPT,
                        inputHint: InputHints.ExpectingInput)
                },
                {
                    ResponseIds.NothingToCancelMessage,
                    (context, data) =>
                    MessageFactory.Text(
                        text: CancelStrings.NOTHING_TO_CANCEL,
                        ssml: CancelStrings.NOTHING_TO_CANCEL,
                        inputHint: InputHints.ExpectingInput)
                },
            }
        };

        public CancelResponses()
        {
            Register(new DictionaryRenderer(_responseTemplates));
        }

        public class ResponseIds
        {
            public const string CancelPrompt = "cancelPrompt";
            public const string CancelConfirmedMessage = "cancelConfirmed";
            public const string CancelDeniedMessage = "cancelDenied";
            public const string NothingToCancelMessage = "nothingToCancel";
        }
    }
}
